x11rb-protocol/src/connection/mod.rs

```diff
@@ -46,6 +54,9 @@ struct SentRequest {
 /// order to keep state for the X11 protocol.
 #[derive(Debug)]
 pub struct Connection {
+    // The setup associated with this connection.
+    setup: Arc<Setup>,
```

Comment:
Correct my if I am wrong, but:

- The whole reason for the `Arc` is so that `Setup` can be kept in both `Connection`s (from x11rb and from x11rb-protocol).
- The one in x11rb-protocol (the place where I am commenting) is unused. You add a getter for it, but nothing actually uses it

If I am right, I would suggest to remove this unused `setup` field. The `from_setup(setup: Arc<Setup>)` function below would work just as well with `setup: &Setup` as its argument.

(Although I still do not think that is even necessary, see that-thread-race comment)